sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -565,6 +565,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
       filters: Array[Filter],
       inputPaths: Array[String],
       broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
And we also need to refresh the partition directory before pruning the partition, probably we need to think more further how to fix that also. In the following PR(s).